src/deployment/deploy.py

```diff
@@ -383,6 +389,68 @@ def create_eventgrid(self):
                 % json.dumps(result.as_dict(), indent=4, sort_keys=True),
             )
 
+    def add_log_export(self):
+        if not self.export_appinsights:
+            logger.info("not exporting appinsights")
+            return
+
+        container_name = "app-insights"
+
+        logger.info("adding appinsight log export")
+        account_name = self.results["deploy"]["func-name"]["value"]
+        key = self.results["deploy"]["func-key"]["value"]
+        account_url = "https://%s.blob.core.windows.net" % account_name
+        client = BlobServiceClient(account_url, credential=key)
+        if container_name not in [x["name"] for x in client.list_containers()]:
+            client.create_container(container_name)
+
+        expiry = datetime.utcnow() + timedelta(days=2 * 365)
```

Comment:
Can we accomplish this with something other than an ad hoc (thus irrevocable) SAS URL?

Reply:
Alternatively: use a much shorter-lived URL, and have a mechanism for refreshing within an instance (though that feels a bit roundabout).

Reply:
The API unfortunately requires SAS urls.

Doing it through the instance requires granting more privileges to the service.

Reply:
Okay. Given the restrictions on perms, and the fact that none of our code touches it (it is only given directly to an Azure API, here), this seems okay to me. Can we just document our usage assumptions? In particular, note that this SAS URL must not "leave this script", except to the extent that it implicitly does in its use in the export request.